### PR TITLE
Add a helper script for testing whether output directories are complete.

### PR DESCRIPTION
**Pull request by tskisner**
Moving some logic out of slurm scripts into a simple python helper script.

### CODE REVIEW

File: sotodlib/toast/scripts/so_batch_control.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+
+def set_obs_state(out_root, obs, state):
+    """Set the current processing state of an observation.
+    """
+    obs_dir = os.path.join(out_root, obs)
+    if not os.path.exists(obs_dir):
+        os.makedirs(obs_dir)
+    state_file = os.path.join(obs_dir, f"state")
+    temp_state = os.path.join(obs_dir, f"state_temp")
```
Comment (keskitalo) [inline sotodlib/toast/scripts/so_batch_control.py:85]: Should the code check if `state_temp` already exists?

File: sotodlib/toast/scripts/so_batch_control.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        None
+
+    """
+    selected = list()
+    for obs in all_obs:
+        state = get_obs_state(out_root, obs)
+        if state is not None:
+            if state == "done":
+                continue
+            if state == "running" and not ignore_running:
```
Comment (keskitalo) [inline sotodlib/toast/scripts/so_batch_control.py:109]: Could we ignore "running" jobs where the file is, say, over 24 hours old?

**Review by keskitalo — APPROVED** [review]
This should be very useful.

File: sotodlib/toast/scripts/so_batch_control.py (hunk 1 above)
Comment (tskisner) [reply sotodlib/toast/scripts/so_batch_control.py]: Although this temp file should only exist for a fraction of a second, I added a check and raise an exception if it exists.

File: sotodlib/toast/scripts/so_batch_control.py (hunk 2 above)
Comment (tskisner) [reply sotodlib/toast/scripts/so_batch_control.py:138]: I added a timeout for this situation

**Pull request merged**